src/erlcloud_kinesis.erl

```diff
@@ -608,28 +608,42 @@ put_record(StreamName, PartitionKey, Data, ExplicitHashKey, Ordering, Options, C
 put_records(StreamName, Items) ->
     put_records(StreamName, Items, default_config()).
 
--spec put_records(binary(), put_records_items(), Config) -> erlcloud_kinesis_impl:json_return() when
-      Config :: aws_config().
+-spec put_records(binary(), put_records_items(), function() | aws_config()) -> erlcloud_kinesis_impl:json_return().
+%% @doc This function takes a fun which is responsible for encoding each of the
+%% data items in the Items list.
+put_records(StreamName, Items, EncodingFun) when is_function(EncodingFun) ->
```

Comment:
passing custom function from outside is fine but stylish comments:
need to export the function spec and probably worth adding arity guard too.

Reply:
+1